### ISSUE TITLE
crashes with "Too many open files"

### ISSUE THREAD

**Issue by Skyfold** [grwlf/haskdogs#6]
It consistently dies on "unordered-containers-0.2.9.0/tests/Strictness.hs" with:

```
> hasktags -c -x STDIN
hasktags: /home/kusanagi/.haskdogs/turtle-1.5.10/src/Turtle/Format.hs: openBinaryFile: resource exhausted (Too many open files)
haskdogs: readCreateProcess: hasktags "-c" "-x" "STDIN" (exit 1): failed
```

I'm not sure if my project has too many dependencies and `Strictness.hs` is the last straw or there is something about that file (or a file right after it) that causes hashktags (called by hashdogs) to crash.

**Comment by sergei-mironov** [comment]
As far as I know, it is a bug of Hasktags. Haskdogs run it and detects this error. I'll update more on this later. Probably, it is fixed in master, we just need to update.

**Comment by sergei-mironov** [comment]
Please check https://github.com/MarcWeber/hasktags/issues/57

**Comment by Skyfold** [comment]
Yep, its just an issue with Hasktags v0.70.0 that is fixed in their v0.71.2. Since this is a recent update its only on Nixpkgs unstable while I was working off of the 18.09 release. Here is my quick fix in my shell.nix:

```
  hasktags = import (pkgs.fetchFromGitHub 
    { owner = "MarcWeber";
      repo = "hasktags";
      inherit 
       ({ url = "https://github.com/MarcWeber/hasktags.git";
          rev = "93d2b29ecf4ac98c7d5f0ebb9f32d9d6e5182e67";
          sha256 = "1brxzia97ffhz9jj53x2ggbw6kyz09pmiijv0s8dv7z5x6rbssf9";
          fetchSubmodules = false;
	}) rev sha256;
    }) {nixpkgs = pkgs; compiler = "ghc843"; };
```

Note: I switched the compiler to "ghc843" instead of "ghc842" since that is not in the 18.09 Nixpkgs release. Also, hasktags forgot to update their nix expression so the revision says v0.70.0 instead of v0.71.2.
